frontend/src/pages/CreateRide.tsx

```diff
@@ -37,43 +36,25 @@ type ValidatableField<T> = {
   invalid: boolean;
 };
 
-export type Ride = {
-  name: string;
-  start: { lat: number; lng: number };
-  end: { lat: number; lng: number };
-  driver?: string;
-  isComplete: boolean;
-  carId?: string;
-  startDate: string;
-  endDate: string;
-};
-
-export type RideRoute = {
-  distance: number;
-  fuelUsed: number;
-  shape: LatLng[];
-};
-
 const createRide = async (ride: Ride, groupId: string, passList?: string[]) => {
-  const newRideRef = await push(ref(db, `${DB_RIDE_COLLECT}`), ride);
-  const rideId = newRideRef.key;
-  await set(ref(db, `${DB_GROUP_COLLECT}/${groupId}/rides/${rideId}`), true);
-  if (passList) {
-    await Promise.all(
-      passList.map(async (p) => {
-        await set(ref(db, `${DB_PASSENGERS_COLLECT}/${rideId}/${p}`), true);
-      })
-    );
+  const rideId = (await setRide(ride)).id;
+  if (rideId) {
+    await setGroupRide(groupId, rideId);
+    createRoute(rideId, ride);
+    if (passList) {
+      await Promise.all(
+        passList.map(async (p) => {
+          setRidePassenger(p, rideId);
```

Comment:
It looks like there was an "async" here before and there isn't now. Is that intentional?